students/sbolsen/lesson05/except_exercise.py

```diff
@@ -0,0 +1,54 @@
+#!/usr/bin/python
+
+"""
+An exercise in playing with Exceptions.
+Make lots of try/except blocks for fun and profit.
+
+Make sure to catch specifically the error you find, rather than all errors.
+"""
+
+from except_test import fun, more_fun, last_fun
+
+
+# Figure out what the exception is, catch it and while still
+# in that catch block, try again with the second item in the list
+
+try:
+    first_try = ['spam', 'cheese', 'mr death']
+    joke = fun(first_try[0])
```

Comment:
This [18] is the only line that should be inside the `try` block here, since that's the function call that raises `NameError`. The list assignment should happen before the try/except, since there's nothing there related to the caught exception, and the call to `fun(first_try[1])` should go in the `except` block after the initial `NameError` is caught.